### PR TITLE
test(duelling-picklist): refactor for cypress tests

### PR DESCRIPTION
**Pull request by stephenogorman**
### Proposed behaviour

Refactor Cypress tests for the `Duelling-Picklist` component to use new cypress-component-react framework for testing.
Add `duelling-picklist.test.js`. 

### Current behaviour

Current process is the regular Cypress approach with BDD.

### Checklist

- [x] Commits follow our style guide
- [ ] Related issues linked in commit messages if required
- [ ] Screenshots are included in the PR if useful
- [ ] All themes are supported if required
- [ ] Unit tests added or updated if required
- [x] Cypress automation tests added or updated if required
- [ ] Storybook added or updated if required
- [ ] Translations added or updated (including creating or amending translation keys table in storybook) if required
- [ ] Typescript `d.ts` file added or updated if required

#### QA

- [x] Tested in CodeSandbox/storybook
- [x] Add new Cypress test coverage if required
- [x] Carbon implementation matches Design System/designs
- [x] UI Tests GitHub check reviewed if required

### Testing instructions

- [x] Run `npx cypress open --component` to check if there is newly added test.file
- [x] Check if the `duelling-picklist.test.js` file passed
- [x] Check all tests were moved/refactored from .feature file -> .test.js file
- [x] Check all unnecessary files were deleted, including .json file
- [x] Run `npx cypress run --component` to check none of the other *.test.js files have regressed
- [x] Run `npx cypress run --e2e` to check none of the feature files have regressed

### CODE REVIEW

**Comment by stephenogorman** [comment]
I asked Daniel Dipper about the following props and he informed me that these are props which are controlled internally, so not for the use of consumers, and not something we can verify - they are covered by unit tests:
highlighted
index
groupIndex
listIndex
isLastItem
isLastGroup

Also, 'Disabled' is listed as a prop for Duelling-Picklist, Picklist and PicklistItem, but it is only used in the Duelling-Picklist component - it probably shouldn't be listed for the others.

File: src/components/duelling-picklist/duelling-picklist.test.js [hunk 1]
```diff
@@ -0,0 +1,990 @@
...
+        <DuellingPicklistComponent rightLabel="Right Label" />
+      );
+
+      picklistRightLabel().should("have.text", "Right Label");
+    });
+  });
+
+  describe("should render Duelling-Picklist component to test Picklist props", () => {
+    it.each(specialCharacters)(
+      "should verify picklist placeholder = %s",
```
Comment (DlgSHi) [inline src/components/duelling-picklist/duelling-picklist.test.js:834]:
```suggestion
      "should verify picklist placeholder is set to %s",
```

File: src/components/duelling-picklist/duelling-picklist.test.js [hunk 2]
```diff
@@ -0,0 +1,990 @@
...
+        }
+        unassignedPicklist().find("div").should("have.text", chars);
+      }
+    );
+
+    it.each([
+      ["locked", true, "have.attr"],
+      ["unlocked", false, "not.have.attr"],
+    ])(
+      "should verify picklist item is %s when locked prop is %s",
```
Comment (DlgSHi) [inline src/components/duelling-picklist/duelling-picklist.test.js:853]: for `locked` state there is also a colour changing, can you add that assertion as well ?

File: src/components/duelling-picklist/duelling-picklist.test.js [hunk 3]
```diff
@@ -0,0 +1,990 @@
...
+          tooltipMessage="Item Locked"
+        />
+      );
+
+      getDataElementByValue("picklist-item").eq(0).children().eq(1).realHover();
+      tooltipPreview().should("have.text", "Item Locked");
+    });
+  });
+
+  describe("should render duelling-picklist with external searchbar and access checkbox", () => {
```
Comment (DlgSHi) [inline src/components/duelling-picklist/duelling-picklist.test.js:879]:
```suggestion
  describe("should render Duelling-Picklist with external searchBar and access checkbox", () => {
```

File: src/components/duelling-picklist/duelling-picklist.test.js [hunk 4]
```diff
@@ -0,0 +1,990 @@
...
+      "should verify picklist search field can be placed outside the Duelling Picklist",
+      (searchString, results) => {
+        CypressMountWithProviders(<DuellingPicklistComponentAlternateSearch />);
+
+        getDataElementByValue("input").type(searchString);
+        unassignedPicklistItems().should("have.length", results);
+      }
+    );
+
+    it("should verify duelling picklist is disabled when access checkox is checked", () => {
```
Comment (DlgSHi) [inline src/components/duelling-picklist/duelling-picklist.test.js:894]:
```suggestion
    it("should verify Duelling Picklist is disabled when access checkbox is checked", () => {
```

File: src/components/duelling-picklist/duelling-picklist.test.js [hunk 5]
```diff
@@ -0,0 +1,990 @@
...
+      CypressMountWithProviders(<DuellingPicklistComponentAlternateSearch />);
+
+      checkBox().check();
+      duellingPicklistComponent().should("have.attr", "disabled");
+    });
+
+    it("should verify duelling picklist is re-enabled when access checkox is unchecked", () => {
```
Comment (DlgSHi) [inline src/components/duelling-picklist/duelling-picklist.test.js:901]:
```suggestion
    it("should verify Duelling Picklist is re-enabled when access checkbox is unchecked", () => {
```

File: src/components/duelling-picklist/duelling-picklist.test.js [hunk 6]
```diff
@@ -0,0 +1,990 @@
...
+      CypressMountWithProviders(<DuellingPicklistComponentAlternateSearch />);
+
+      checkBox().check();
+      duellingPicklistComponent().should("have.attr", "disabled");
+      checkBox().uncheck();
+      duellingPicklistComponent().should("not.have.attr", "disabled");
+    });
+  });
+
+  describe("should render duelling-picklist with items grouped and a picklist divider", () => {
```
Comment (DlgSHi) [inline src/components/duelling-picklist/duelling-picklist.test.js:911]:
```suggestion
  describe("should render Duelling Picklist with items grouped and a picklist divider", () => {
```

File: src/components/duelling-picklist/duelling-picklist.test.js [hunk 7]
```diff
@@ -0,0 +1,990 @@
...
+          {...props}
+        >
+          {renderItems(selectedItems, "remove", onRemove)}
+        </Picklist>
+      </DuellingPicklist>
+    </div>
+  );
+};
+
+context("Testing Duelling-Picklist component", () => {
```
Comment (DlgSHi) [inline src/components/duelling-picklist/duelling-picklist.test.js:679]: we should use: 
`Duelling-Picklist` or `Duelling Picklist` - your choice :) 

```suggestion
context("Testing Duelling-Picklist component", () => {
```

File: src/components/duelling-picklist/duelling-picklist.test.js [hunk 8]
```diff
@@ -0,0 +1,990 @@
...
+    });
+
+    it("should verify all items in a group are added to assigned picklist when group add button is clicked", () => {
+      CypressMountWithProviders(<DuellingPicklistComponentGrouped />);
+
+      getDataElementByValue("picklist-group").children().eq(1).click();
+      assignedPicklistItems().should("have.length", "3");
+      getDataElementByValue("picklist")
+        .eq(1)
+        .find('[data-element="picklist-group"]')
```
Comment (DlgSHi) [inline src/components/duelling-picklist/duelling-picklist.test.js:934]: can we put this locator in `locators.js` file?

File: src/components/duelling-picklist/duelling-picklist.test.js [hunk 9]
```diff
@@ -0,0 +1,990 @@
...
+    beforeEach(() => {
+      callback = cy.stub();
+    });
+
+    it("should call onChange when add button clicked", () => {
+      CypressMountWithProviders(
+        <DuellingPicklistComponentPicklistProps onChange={callback} />
+      );
+
+      addButton(0)
```
Comment (DlgSHi) [inline src/components/duelling-picklist/duelling-picklist.test.js:964]: what about `removeButton`?

**Review by DlgSHi — CHANGES_REQUESTED** [review]

**Review by DlgSHi — CHANGES_REQUESTED** [review]
I think you missed few props to cover :)

**Review by DlgSHi — APPROVED** [review]

**Review by divyajindel — APPROVED** [review]

**Review by DianeGit22 — APPROVED** [review]

**Pull request merged**